pkg/cluster/streams.go

```diff
@@ -0,0 +1,261 @@
+			streamSink := getEventStreamSink(stream, eventType)
+
+			eventStreams = append(eventStreams, zalandov1alpha1.EventStream{
+				EventStreamFlow:   streamFlow,
+				EventStreamSink:   streamSink,
+				EventStreamSource: streamSource})
+		}
+	}
+
+	return &zalandov1alpha1.FabricEventStream{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       constants.EventStreamSourceCRDKind,
+			APIVersion: "zalando.org/v1alphav1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        c.Name,
+			Namespace:   c.Namespace,
+			Annotations: c.AnnotationsToPropagate(c.annotationsSet(nil)),
+		},
+		Spec: zalandov1alpha1.FabricEventStreamSpec{
+			ApplicationId: "",
+			EventStreams:  eventStreams,
+		},
+	}
+}
+
+func (c *Cluster) getEventStreamSource(stream acidv1.Stream, table, eventType string) zalandov1alpha1.EventStreamSource {
+	_, schema := getTableSchema(table)
+	streamFilter := stream.Filter[table]
+	return zalandov1alpha1.EventStreamSource{
+		Type:             constants.EventStreamSourcePGType,
+		Schema:           schema,
+		EventStreamTable: getOutboxTable(table, eventType),
+		Filter:           streamFilter,
+		Connection:       c.getStreamConnection(stream.Database, constants.EventStreamSourceSlotPrefix+constants.UserRoleNameSuffix),
+	}
+}
+
+func getEventStreamFlow(stream acidv1.Stream) zalandov1alpha1.EventStreamFlow {
+	return zalandov1alpha1.EventStreamFlow{
+		Type: constants.EventStreamFlowPgGenericType,
+	}
+}
+
+func getEventStreamSink(stream acidv1.Stream, eventType string) zalandov1alpha1.EventStreamSink {
+	return zalandov1alpha1.EventStreamSink{
+		Type:         constants.EventStreamSinkNakadiType,
+		EventType:    eventType,
+		MaxBatchSize: stream.BatchSize,
+	}
+}
+
+func getTableSchema(fullTableName string) (tableName, schemaName string) {
+	schemaName = "public"
+	tableName = fullTableName
+	if strings.Contains(fullTableName, ".") {
+		schemaName = strings.Split(fullTableName, ".")[0]
+		tableName = strings.Split(fullTableName, ".")[1]
+	}
+
+	return tableName, schemaName
+}
+
+func getOutboxTable(tableName, eventType string) zalandov1alpha1.EventStreamTable {
+	return zalandov1alpha1.EventStreamTable{
+		Name: outboxTableNameTemplate.Format("table", tableName, "eventtype", eventType),
+	}
+}
+
+func (c *Cluster) getStreamConnection(database, user string) zalandov1alpha1.Connection {
+	return zalandov1alpha1.Connection{
+		Url:      fmt.Sprintf("jdbc:postgresql://%s.%s/%s?user=%s&ssl=true&sslmode=require", c.Name, c.Namespace, database, user),
+		SlotName: c.getLogicalReplicationSlot(database),
+		DBAuth: zalandov1alpha1.DBAuth{
+			Type:        constants.EventStreamSourceAuthType,
+			Name:        c.credentialSecretNameForCluster(user, c.Name),
+			UserKey:     "username",
+			PasswordKey: "password",
+		},
+	}
+}
+
+func (c *Cluster) getLogicalReplicationSlot(database string) string {
+	for slotName, slot := range c.Spec.Patroni.Slots {
+		if slot["type"] == "logical" && slot["database"] == database {
```

Comment:
what if there are logical slots unrelated to CDC?

Reply:
in this case, it would be referenced here. Maybe I can also check for the wal2json plugin. Hm or better rely on a fixed naming schema...